client/src/api/request/index.js

```diff
@@ -0,0 +1,28 @@
+import jsonFetch from "simple-json-fetch";
+
+/**
+ * Request
+ * @param  {string} requestUrl url end_point
+ * @param  {object} requestParameters
+ * @param  {func} callback
+ */
+const request = (requestUrl = null, requestParameters, callback) => {
+  if (null == requestUrl) {
+    throw new Error(`url ${requestUrl} malformed `);
+  }
+  if ("function" != typeof callback) {
+    throw new Error(`callback - ${callback} - is not a function`);
+  }
+  return dispatch => {
+    return jsonFetch(requestUrl, requestParameters).then(
+      response => {
+        dispatch(callback(response, requestParameters));
+      },
+      response => {
```

Comment:
(im obviously not familiar with this module... can you explain why we dispatch the callback twice?)